### PR TITLE
Update multiBamSummary: label option

### PR DESCRIPTION
**Pull request by gallardoalba**
**Welcome to deepTools GitHub repository! Please check the following regarding
your pull request :**

 - [ ] Does the PR contain new feature?
 - [x] Does the PR contain bugfix?
 - [ ] Does the PR contain documentation changes?
 - [ ] Does the PR contain changes to the galaxy wrapper?

### CODE REVIEW

File: galaxy/wrapper/multiBamSummary.xml [hunk 1]
```diff
@@ -17,7 +17,10 @@
 
             --outFileName '$outFile'
             --bamfiles #echo " ".join($files)#
-            --labels #echo " ".join($labels)#
+
+            #if $custom_sample_labels_conditional.custom_labels_select == 'Yes'
+                --labels #echo ' '.join($custom_sample_labels_conditional.labels)#
```
[the remark this answers is not in the document]
Comment (gallardoalba) [reply galaxy/wrapper/multiBamSummary.xml:22]: It seems to be sanitized if I'm not wrong https://github.com/gallardoalba/deepTools/blob/Fix_label/galaxy/wrapper/deepTools_macros.xml#L446, and about quoting not sure what do you mean sorry.

[the remark this answers is not in the document]
Comment (mvdbeek) [reply galaxy/wrapper/multiBamSummary.xml:22]: That is not enough, `;` is part of the printable set, so you could basically inject arbitrary commands, and there are no quotes here at all.

[the remark this answers is not in the document]
Comment (mvdbeek) [reply galaxy/wrapper/multiBamSummary.xml:22]: https://github.com/deeptools/deepTools/blob/master/galaxy/wrapper/computeMatrix.xml#L27 has the same issue btw, i think it's worth grepping for this pattern to make sure we fix all cases

[the remark this answers is not in the document]
Comment (mvdbeek) [reply galaxy/wrapper/multiBamSummary.xml:22]: https://github.com/deeptools/deepTools/blob/master/galaxy/wrapper/deepTools_macros.xml#L501 is how I think the quoting should happen (for when you don't have custom labels), it just needs to be applied here as well.
Or, you could refactor this a little, something like
```
        #set files=[]
        #set labels=[]
        #import re
        #if $multibam_conditional.orderMatters == "No":
            #for $counter, $bamfile in enumerate($multibam_conditional.bamfiles):
                #set identifier = re.sub('[^\.\s\w\-]', '_', str($bamfile.element_identifier))
                ln -s '${bamfile}' './${counter}.bam' &&
                #if $bamfile.ext == 'bam':
                    ln -s '${bamfile.metadata.bam_index}' './${counter}.bam.bai' &&
                #else:
                    ln -s '${bamfile.metadata.cram_index}' './${counter}.bam.crai' &&
                #end if
                #silent $files.append("'%s.bam'" % $counter)
                #silent $labels.append($identifier)
            #end for
        #else:
            #for $counter, $f in enumerate($multibam_conditional.multibam_repeats):
                #set identifier = re.sub('[^\.\s\w\-]', '_', str($f.bamfiles.element_identifier))
                ln -s '${f.bamfiles}' './${counter}.bam' &&
                #if $f.bamfiles.ext == 'bam':
                    ln -s '${f.bamfiles.metadata.bam_index}' './${counter}.bam.bai' &&
                #else:
                    ln -s '${f.bamfiles.metadata.cram_index}' './${counter}.bam.crai' &&
                #end if
                #silent $files.append("'%s.bam'" % $counter)
                #silent $labels.append($identifier)
            #end for
        #end if
        #if $custom_sample_labels_conditional.custom_labels_select == "Yes":
            #set labels=custom_labels
        #end if
        #silent labels=["'%s'" % label for label in labels]
```

[the remark this answers is not in the document]
Comment (gallardoalba) [reply galaxy/wrapper/multiBamSummary.xml:22]: Perfect, thanks a lot!

[the remark this answers is not in the document]
Comment (gallardoalba) [reply galaxy/wrapper/multiBamSummary.xml]: > https://github.com/deeptools/deepTools/blob/master/galaxy/wrapper/computeMatrix.xml#L27 has the same issue btw, i think it's worth grepping for this pattern to make sure we fix all cases

Sorry, but after reviewing I still don't understand what you mean. For example in that case that you are indicating here, it makes use of https://github.com/deeptools/deepTools/blob/master/galaxy/wrapper/deepTools_macros.xml#L522 for storing the labels.

**Comment by bgruening** [comment]
> Job in error state.. tool_id: deeptools_multi_bam_summary, exit_code: 2, stderr: usage: multiBamSummary bins --bamfiles file1.bam file2.bam -o results.npz 
multiBamSummary bins: error: argument --labels/-l: expected at least one argument

@gallardoalba ping

**Pull request closed without merging**